stanford-earth/SE3_BLT
# Add test environment as an external package

# DO NOT MERGE
- Moves several Behat related packages that BLT may need, from se3_blt's composer.json to drupal-caravan's.

# User stories

- As a developer I want to run all of the behat tests for my site on my local environment.
- It would be nice to have the tests live in the module's repo as well.
- As a developer I want to always have the appropriate version of behat tests for each module in my project.

# Summary
Config settings and scripts for "easily" building a docker container in which to run Behat tests.  This pull request should allow someone to run tests on their local machines.  It will also run and test updates to this build repository in Travis.

The majority of code to be reviewed, can be found in this repository ([repo](https://github.com/kbrownell/drupal-caravan)).

Further documentation, including a list of outstanding issues ([doc](https://docs.google.com/document/d/1DAoQHLwqRwjRmY-_oCjhcN1VJAritX-aaqXEK_ooMj8/edit#)).

# Steps to test
**WARNING:** This will append values to local config files, including blt/project.local.yml and tests/behat/behat.local.yml.

**Requirements:** You will need to have Docker Edge installed ([download for Mac](https://docs.docker.com/docker-for-mac/install/)).

You will also need an ssh key pair named, `acquia_id_rsa` that is also saved to your Acquia account.

If you are on a Mac, and want to access the container's site through your browser, you'll need to run:

`sudo ifconfig lo0 alias 192.168.88.88/24`

1. Checkout drupal-caravan branch
2. Run `composer update`
3. Run `vendor/kbrownell/drupal-caravan/setup-caravan.sh`
4. After ~20m of setup, this should run two simple behat test

# Tickets
There are a number of tickets related to this project, but we could use: [EARTH-253](https://stanfordits.atlassian.net/projects/EARTH/issues/EARTH-253) for this particular PR.
